agent/verify_job.go

```diff
@@ -0,0 +1,95 @@
+package agent
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/buildkite/agent/v3/internal/pipeline"
+)
+
+var ErrNoSignature = errors.New("job had no signature to verify")
+
+type invalidSignatureError struct {
```

Comment:
When we get to signature schemes that separate message digest and signature, it might be helpful (for the poor saps diagnosing signing failures) to separate "digest didn't match message" and "the digest matched the message but the signature isn't valid for the digest" - both cases seem like "invalidSignatureError" though so I don't know if I'm arguing for any change here.

Reply:
I guess the go way is to let that be handled by unwrapping the underlying error.